apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx

```diff
@@ -600,6 +612,21 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
         }
     };
 
+    const [ selectedAuthMethod, setSelectedAuthMethod ] = useState<string>(undefined);
+
+    const handleAuthMethodChange = (values: Map<string, FormValue>) => {
+        const authMethod: string = values.get("tokenEndpointAuthMethod") as string;
+
+        setSelectedAuthMethod(authMethod);
+    };
+
+    useEffect(() => {
+
+        if (!selectedAuthMethod) {
+            setSelectedAuthMethod(initialValues?.clientAuthentication?.tokenEndpointAuthMethod);
+        }
+    });
```

Comment:
Consider defining the dependency array for this useEffect. Otherwise, this will run after every render of the component.

Reply:
fixed with [be309bd](https://github.com/wso2/identity-apps/pull/4284/commits/be309bdd61f1ecdcf606cdf0bfe85ea775a210a6)